edx_django_utils/monitoring/docs/how_tos/using_custom_metrics.rst

```diff
@@ -0,0 +1,144 @@
+Enhanced Monitoring and Custom Metrics
+======================================
+
+.. contents::
+   :local:
+   :depth: 2
+
+What is a Custom Metrics?
+-------------------------
+
+A custom metric is a name/value pair that you define, which will be associated with a request and available in various places in your monitoring software. For example, in Open edX, there is a ``request_user_id`` made available to all requests using the `RequestMetricsMiddleware`_.
+
+If you are using NewRelic, you can search for PageVeiws, Transactions, TransactionErrors, etc. using NewRelic Insights, either with custom queries using the Data Explorer (searching "Events", not Metrics). They will also appear in other places, like when viewing an error in NewRelic APM.
+
+.. _RequestMetricsMiddleware: https://github.com/edx/edx-drf-extensions/blob/master/edx_rest_framework_extensions/middleware.py#L12-L39
+
+Coding New Custom Metrics
+-------------------------
+
+After setting up Middleware as detailed in README.rst, adding a new custom metric is as simple as::
+
+    from openedx.core.djangoapps import monitoring_utils
+
+    monitoring_utils.set_custom_metric(name, value)
+
+* Supported values are strings, bools and numbers.
+* Earlier values will be overwritten if the same name is set multiple times during a request.
+
+The following methods only work with number values, and do *not* overwrite earlier values, but instead make use of earlier values within a request::
+
+    monitoring_utils.accumulate(name, value)
+    monitoring_utils.increment(name, value)
+
+For a complete list of the public methods available, see the ``__init__.py`` file and the docstrings for details.
+
+If you require functionality from ``newrelic.agent`` that hasn't yet been abstracted, please add any additional functionality to keep it encapsulated.
+
+Tips for Using Custom Metrics
+-----------------------------
+
+* A single name with an enum-like string value is often better than a set of related names with boolean values. For example:
+
+  * **DO**:
+
+    * request_auth_type: unauthenticated/jwt-cookie/session
+
+  * **DON'T** do:
+
+    * is_request_unauthenticated: True/False,
+    * is_request_auth_jwt_cookie: True/False,
+    * is_request_auth_session: True/False.
+
+* When possible, set the custom metric for both the positive and negative case. This helps avoid misinterpreting an incorrect query that doesn't turn up a metric.
+* Easily check a custom metric across multiple environments (``appName`` in NewRelic).  For example, for edx.org, this means checking Production and Edge, where Edge might often be forgotten.
```

Comment:
I think this first sentence might be incomplete? I'm not sure what it's trying to say.

Reply:
@timmc-edx: This was more of a deprecation tip, so I moved it and tried to add clarity.  Let me know what you think.

Reply:
I don't see the change—did you push it yet?

Reply:
`git push` to the rescue.  Sorry about that. ;)

Reply:
Cool, looks good!